src/tensorflow.cc

```diff
@@ -1280,31 +1273,36 @@ AutoCompleteHelper::FixBatchingSupport()
               } else {
                 config_io.MemberAsArray("dims", &config_dims);
               }
-              if (config_dims.ArraySize() != 0) {
-                // look up corresponding io info from model
-                for (const TRITONTF_IOList* itr = model_ios[ios_idx];
-                     itr != nullptr; itr = itr->next_) {
-                  TRITONTF_IO* io = itr->io_;
-                  if (config_name == io->name_) {
-                    bool should_batch =
-                        (io->shape_->rank_ == (config_dims.ArraySize() + 1));
-                    // inconsistent hint
-                    if (config_batch_hint &&
-                        (model_support_batching_ != should_batch)) {
-                      return TRITONSERVER_ErrorNew(
-                          TRITONSERVER_ERROR_INTERNAL,
-                          std::string(
-                              "unable to autofill for '" +
-                              model_state_->Name() +
-                              "', model tensor configurations are "
-                              "contradicting " +
-                              "each other in terms of whether batching is "
-                              "supported")
-                              .c_str());
-                    }
-                    config_batch_hint = true;
-                    model_support_batching_ = should_batch;
+              for (const TRITONTF_IOList* itr = model_ios[ios_idx];
+                   itr != nullptr; itr = itr->next_) {
+                TRITONTF_IO* io = itr->io_;
+                if (config_name == io->name_) {
+                  bool model_io_explicit = io->shape_->rank_ > 0;
+                  bool model_shape_matches_config_shape =
+                      io->shape_->rank_ == config_dims.ArraySize() + 1
```

Comment:
I think you missed ";"
Re-run the formatter. This code should not ideally even compile.

Reply:
Context switching is hard. Fixed.